tools/data_source/upload.py

```diff
@@ -5,6 +5,7 @@
 # to be reflected in galaxy.web.controllers.tool_runner and galaxy.tools
 from __future__ import print_function
 
+import logging
```

Comment:
Your comment is noted, the "import" statement was committed by unintentional. I have removed the line, but your input is noted.